tests/integration_tests/platforms.py

```diff
@@ -0,0 +1,168 @@
+                self.settings.EXISTING_INSTANCE_ID
+            )
+            return
+        image_id = self._get_image()
+        launch_args = {
+            'image_id': image_id,
+            'user_data': self.user_data,
+            'wait': self.wait,
+        }
+        if self.instance_type:
+            launch_args['instance_type'] = self.instance_type
+        launch_args.update(self.launch_kwargs)
+        self.instance = self.client.launch(**launch_args)
+        print('Launched instance: {}'.format(self.instance))
+
+    def destroy(self):
+        self.instance.delete()
+
+    def exec(self, command):
+        return self.instance.execute(command)
+
+    def get_file(self, remote_file, local_file=None):
+        if not local_file:
+            local_file = StringIO()
+        self.instance.pull_file(remote_file, local_file)
+        return local_file
+
+    def put_file(self, local_path, remote_path):
+        self.instance.push_file(local_path, remote_path)
+
+    def snapshot(self):
+        return self.client.snapshot(self.instance, clean=True, wait=True)
+
+    def generate_proposed_image(self, sudo=False):
+        print('Generating proposed image')
+        remote_script = (
+            '{sudo} echo deb "http://archive.ubuntu.com/ubuntu '
+            '$(lsb_release -sc)-proposed main" | '
+            '{sudo} tee /etc/apt/sources.list.d/proposed.list\n'
+            '{sudo} apt-get update -q\n'
+            '{sudo} apt-get install -qy cloud-init'
+        ).format(sudo='sudo' if sudo else '')
+        self.exec(remote_script)
+        version = self.exec('cloud-init -v').split()[-1]
+        print('Installed cloud-init version: {}'.format(version))
+        self.instance.clean()
+        image_id = self.snapshot()
+        print('Created proposed image: {}'.format(image_id))
+        set_current_image(image_id)
+
+    def __enter__(self):
+        self.launch()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        if not self.settings.KEEP_INSTANCE:
+            self.destroy()
+
+
+class OracleClient(IntegrationClient):
+    datasource = 'oracle'
+
+    def __init__(self, user_data=None, instance_type=None, wait=True,
+                 settings=integration_settings, launch_kwargs=None):
+        super(OracleClient, self).__init__(
+            user_data,
+            instance_type,
+            wait,
+            settings,
+            launch_kwargs
+        )
+        self.client = OCI(
+            tag='OCI Integration Test',
+            compartment_id=self.settings.ORACLE_COMPARTMENT_ID
+        )
+
+    def generate_proposed_image(self, sudo=True):
+        super(OracleClient, self).generate_proposed_image(sudo)
+
+
+class LxdContainerClient(IntegrationClient):
+    datasource = 'lxd_container'
+
+    def __init__(self, user_data=None, instance_type=None, wait=True,
+                 settings=integration_settings, launch_kwargs=None):
+        super(LxdContainerClient, self).__init__(
```

Comment:
I think cloud-init only support python3 right now, right ? If that is the case, maybe we can use the less verbose `super` call from python3